components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java

```diff
@@ -167,8 +167,8 @@ public static APIDTO fromAPItoDTO(API model, String tenantDomain) throws APIMana
         Set<org.wso2.carbon.apimgt.api.model.Tier> apiTiers = model.getAvailableTiers();
         List<APITiersDTO> tiersToReturn = new ArrayList<>();
         int tenantId = 0;
-        if (!StringUtils.isBlank(tenantDomain)) {
-            tenantId = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+        if (!StringUtils.isBlank(organization)) {
+            tenantId = APIUtil.getTenantIdFromTenantDomain(organization);
```

Comment:
This should be changed to getInternalOrganizationId(organization);

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/448f1494c616bef5b514c816f3a29c6dda74a465